include/deal.II/base/standard_tensors.h

```diff
@@ -0,0 +1,513 @@
+ * @f]
+ * such that
+ * @f[
+ * 	\dfrac{1}{2}[\mathbf{A} + \mathbf{A}^T] = \mathcal{S}:\mathbf{A}
+ *    \qquad \text{and} \qquad
+ * 	\dfrac{1}{2}[\mathbf{A} - \mathbf{A}^T] = \mathcal{W}:\mathbf{A} \, .
+ * @f]
+ * We can also define a tensor $\mathcal{P}$ that performs the projection
+ * operation
+ * @f[
+ * 	\mathbf{A} - [\frac{1}{dim} \textrm{trace}(\mathbf{A})] \mathbf{I}
+ *  = \mathcal{P}:\mathbf{A} \, .
+ * @f]
+ * The fourth-order SymmetricTensor returned by identity_tensor() is
+ * $\mathcal{S}$, and that returned by deviator_tensor() is $\mathcal{P}$.
+ *
+ * For convenience we will predefine come commonly referenced tensors and
+ * operations.
+ * Using the position vector $\mathbf{X}$ in the referential (material)
+ * configuration, the displacement vector is given as
+ * $\mathbf{u} = \mathbf{u}(\mathbf{X})$.
+ * From this we can compute the deformation gradient tensor as
+ * @f[
+ *  \mathbf{F} := \mathbf{I} + \nabla_{0}\mathbf{u} \, ,
+ * @f]
+ * wherein the differential operator $\nabla_{0}$ is defined as
+ * $\frac{\partial}{\partial \mathbf{X}}$.
+ * Finally, two common tensor operators are represented by $\cdot$ and $:$
+ * operators. These respectively represent a single and double contraction over
+ * the inner tensor indices.
+ *
+ * @relates Tensor
+ * @relates SymmetricTensor
+ * @author Jean-Paul Pelteret, Andrew McBride, 2016
+*/
+template <int dim>
+class Elasticity
+{
+public:
+
+  /**
+   * The second-order referential/spatial symmetric identity (metric) tensor
+   * $\mathbf{I}$.
+   *
+   * This is defined such that, for any second-order Tensor or SymmetricTensor,
+   * the following holds:
+   * @f[
+   *  \mathbf{I} \cdot \{ \bullet \} = \{ \bullet \} \cdot \mathbf{I} =
+   * \{ \bullet \}
+   *    \qquad \text{and} \qquad
+   * \mathbf{I} : \{ \bullet \} = \textrm{trace} \{ \bullet \} \, .
+   * @f]
+   * This definition aligns with the second-order SymmetricTensor returned by
+   * unit_symmetric_tensor().
+   */
+  static const SymmetricTensor<2, dim> I;
+
+  /**
+   * Returns the deformation gradient Tensor,
+   * as constructed from the material displacement gradient @p Grad_u.
+   * The result is expressed as
+   * @f[
+   *  \mathbf{F} := \mathbf{I} + \nabla_{0}\mathbf{u}
+   * @f]
+   * where $\mathbf{u} = \mathbf{u}(\mathbf{X})$ is the displacement at position
+   * $\mathbf{X}$ in the referential configuration.
+   * The differential operator $\nabla_{0}$ is defined as
+   * $\frac{\partial}{\partial \mathbf{X}}$.
+   */
+  template <typename Number>
+  static Tensor<2, dim, Number>
+  F (const Tensor<2, dim, Number> &Grad_u);
+
+  /**
+   * Returns the right Cauchy-Green SymmetricTensor,
+   * as constructed from the deformation gradient tensor @p F.
+   * The result is expressed as
+   * @f[
+   *  \mathbf{C} := \mathbf{F}^{T}\cdot\mathbf{F} \, .
+   * @f]
+   */
+  template <typename Number>
+  static SymmetricTensor<2, dim, Number>
+  C (const Tensor<2, dim, Number> &F);
```

Comment:
In the example below where you use `C`, it struck me that for it to make sense, `C(F)` necessarily had to have no units. I think that's actually true here. But I think it would make sense for each of these functions to state what the units of all quantities are. (Here, of `F` and of what the function returns.) We've found in ASPECT that diligently documenting all units of everything is a really useful process in finding subtle bugs.